apps/dashboard/app/models/batch_connect/session_context.rb

```diff
@@ -59,6 +59,37 @@ def method_missing(method_name, *arguments, &block)
     # @return [Boolean]
     def respond_to_missing?(method_name, include_private = false)
       /^(?<id>[^=]+)=?$/ =~ method_name.to_s && self[id] || super
+   end
+    
+
+    # @return [Boolean]
+    def attribute_cache_enabled?
+      self.any? {|v| v.opts[:cacheable]  } 
+    end
+    
+    
+    # @return [Boolean]   
+    def app_specific_cache_enabled?
+      self[:cacheable]
+    end
+   
+     # @return [Boolean]   
+    def global_cache_enabled? 
+       ENV["OOD_BATCH_CONNECT_CACHE_ATTR_VALUES"]
     end
+
+    #Logic determining if attributes should be pulled from cache
+    def update_with_cache(cache, app_specific_cache)
```

Comment:
The second argument `app_specific_cache` is not used and should be removed. Storing app_specific_cache_setting information on the session_context like you are doing makes the most sense because the app is what has the build method to create the session context

Reply:
I guess the problem I ran into here is that the form_config is a private method on app, the model has no way of accessing this, unless the controller shares it with the model correct ? Or maybe in the build_session_context in app.rb attach app_specific_cache_setting to the session_context model ?